Baby Monitor/Source Files/Services/Connection/WebRTC/WebRtcVideoCapturer.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  WebRtcVideoCapturer.swift
+//  Baby Monitor
+//
+
+import WebRTC
+
+protocol VideoCapturer {
+    var isCapturing: Bool { get }
+    func startCapturing()
+    func stopCapturing()
+}
+
+final class WebRTCVideoCapturer: VideoCapturer {
+
+    private(set) var isCapturing = false
+    private let device: AVCaptureDevice
+    private let format: AVCaptureDevice.Format
+    private let framesPerSecond: Int
+    private let capturer: RTCCameraVideoCapturer
+
+    init(device: AVCaptureDevice, format: AVCaptureDevice.Format, framesPerSecond: Int, capturer: RTCCameraVideoCapturer) {
+        self.device = device
+        self.format = format
+        self.framesPerSecond = framesPerSecond
+        self.capturer = capturer
+    }
+
+    func startCapturing() {
+        capturer.startCapture(with: device, format: format, fps: framesPerSecond) { [weak self] _ in
+            self?.isCapturing = true
```

Comment:
This is probably the place where non-optional `error` is passed into the closure ;)
I'd either try to convert it to optional `NSError` or check if `code` is != 0

Reply:
Actually, it's error! underhood, so any call to it can result in crash. I will add a comment about that here.

Reply:
yeah, just leave it then. Unless there is a safe way to unpack it, we need to assume capturing session started when a callback is called ;)